cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl.hpp

```diff
@@ -0,0 +1,359 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/table/common.hpp"
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+#include "oneapi/dal/backend/primitives/utils.hpp"
+#include "oneapi/dal/algo/decision_forest/train_types.hpp"
+
+#include "oneapi/dal/backend/primitives/rng/rng_engine.hpp"
+
+#include "oneapi/dal/algo/decision_forest/backend/gpu/train_misc_structs.hpp"
+#include "oneapi/dal/algo/decision_forest/backend/gpu/train_impurity_data.hpp"
+#include "oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels.hpp"
+#include "oneapi/dal/algo/decision_forest/backend/gpu/train_feature_type.hpp"
+#include "oneapi/dal/algo/decision_forest/backend/gpu/train_model_manager.hpp"
+#include "oneapi/dal/algo/decision_forest/backend/gpu/train_best_split_impl.hpp"
+
+#ifdef DISTRIBUTED_SUPPORT_ENABLED
+#include "oneapi/dal/policy/mpi.hpp"
+#endif
+
+namespace oneapi::dal::decision_forest::backend {
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+namespace de = dal::detail;
+namespace bk = dal::backend;
```

Comment:
"bk" and "be" in different files. Maybe stick to "bk" - it is more widely used in oneDAL?